packages/react-form/src/hooks/test/form.test.tsx

```diff
@@ -200,51 +205,58 @@ describe('useForm', () => {
 
       wrapper.find(TextField, {label: 'title'})!.trigger('onChange', '');
 
-      wrapper
-        .find('button', {type: 'submit'})!
-        .trigger('onClick', clickEvent());
+      wrapper.act(() => {
+        wrapper
+          .find('button', {type: 'submit'})!
+          .trigger('onClick', clickEvent());
+      });
 
       expect(submitSpy).not.toHaveBeenCalled();
       expect(wrapper).toContainReactComponent('p', {
         children: 'Title is required!',
       });
     });
 
-    // Async act() issues block this test https://github.com/facebook/react/issues/15379
-    // eslint-disable-next-line jest/no-disabled-tests
-    it.skip('returns remote submission errors', async () => {
+    it('returns remote submission errors', async () => {
       const error = {message: 'The server hates it'};
-      const submitSpy = jest.fn(() => Promise.resolve(submitFail([error])));
+      const promise = Promise.resolve(submitFail([error]));
+
       const wrapper = mount(
-        <ProductForm data={fakeProduct()} onSubmit={submitSpy} />,
+        <ProductForm data={fakeProduct()} onSubmit={() => promise} />,
       );
 
-      await wrapper
-        .find('button', {type: 'submit'})!
-        .trigger('onClick', clickEvent());
+      await wrapper.act(async () => {
+        await wrapper
```

Comment:
Seems weird that sometimes you are waiting on calling this, sometimes you are waiting on the promise. Stick to one.